File: backend/MedicalHistory.Application/Services/DiseaseService.cs
using MedicalHistory.Core.Interfaces.Repositories;
using MedicalHistory.Core.Interfaces.Services;
using MedicalHistory.Core.Models;

namespace MedicalHistory.Application.Services;

public class DiseaseService : IDiseaseService
{
    private readonly IDiseaseRepository _diseaseRepository;

    public DiseaseService(IDiseaseRepository diseaseRepository)
    {
        _diseaseRepository = diseaseRepository;
    }

    public async Task<Guid> CreateDisease(Disease disease)
    {
        return await _diseaseRepository.Create(disease);
    }

    public async Task<Guid> UpdateDisease(Disease disease)
    {
        return await _diseaseRepository.Update(disease);
    }

    public async Task<Guid> DeleteDisease(Guid id)
    {
        return await _diseaseRepository.Delete(id);
    }

    public async Task<Disease> GetDisease(Guid id)
    {
        var disease = await _diseaseRepository.Get(id);
        return disease;
    }

    public async Task<List<Disease>> GetAllDiseases()
    {
        return await _diseaseRepository.GetAll();
    }

}

File: frontend/src/services/diseaseService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/disease';

class DiseaseService {
  async getAllDiseases() {
    const response = await axios.get(API_BASE_URL);
    return response.data;
  }

  async getDiseaseById(id) {
    const response = await axios.get(`${API_BASE_URL}/${id}`);
    return response.data;
  }

  async createDisease(diseaseData) {
    const response = await axios.post(API_BASE_URL, diseaseData);
    return response.data;
  }

  async updateDisease(id, diseaseData) {
    const response = await axios.put(`${API_BASE_URL}/${id}`, diseaseData);
    return response.data;
  }

  async deleteDisease(id) {
    const response = await axios.delete(`${API_BASE_URL}/${id}`);
    return response.data;
  }
}

export default new DiseaseService();